src/fidesops/api/v1/endpoints/drp_endpoints.py

```diff
@@ -0,0 +1,154 @@
+from fidesops.service.privacy_request.request_runner_service import PrivacyRequestRunner
+from fidesops.service.privacy_request.request_service import (
+    build_required_privacy_request_kwargs,
+    cache_data,
+)
+from fidesops.util.cache import FidesopsRedis
+from fidesops.util.oauth_util import verify_oauth_client
+
+logger = logging.getLogger(__name__)
+router = APIRouter(tags=["DRP"], prefix=urls.V1_URL_PREFIX)
+EMBEDDED_EXECUTION_LOG_LIMIT = 50
+
+
+@router.post(
+    urls.DRP_EXERCISE,
+    status_code=HTTP_200_OK,
+    response_model=PrivacyRequestDRPStatusResponse,
+)
+def create_drp_privacy_request(
+    *,
+    cache: FidesopsRedis = Depends(deps.get_cache),
+    db: Session = Depends(deps.get_db),
+    data: DrpPrivacyRequestCreate,
+) -> PrivacyRequestDRPStatusResponse:
+    """
+    Given a drp privacy request body, create and execute
+    a corresponding Fidesops PrivacyRequest
+    """
+
+    jwt_key: str = config.security.DRP_JWT_SECRET
+    if jwt_key is None:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="JWT key must be provided",
+        )
+
+    logger.info(f"Finding policy with drp action '{data.exercise[0]}'")
+    policy: Optional[Policy] = Policy.get_by(
+        db=db,
+        field="drp_action",
+        value=data.exercise[0],
+    )
+
+    if not policy:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No policy found with drp action '{data.exercise}'.",
+        )
+
+    privacy_request_kwargs: Dict[str, Any] = build_required_privacy_request_kwargs(
+        None, policy.id
+    )
+
+    try:
+        privacy_request: PrivacyRequest = PrivacyRequest.create(
+            db=db, data=privacy_request_kwargs
+        )
+
+        logger.info(f"Decrypting identity for DRP privacy request {privacy_request.id}")
+
+        decrypted_identity: DrpIdentity = DrpIdentity(
+            **jwt.decode(data.identity, jwt_key, algorithms=["HS256"])
+        )
+
+        mapped_identity: PrivacyRequestIdentity = DrpFidesopsMapper.map_identity(
+            drp_identity=decrypted_identity
+        )
+
+        cache_data(privacy_request, policy, mapped_identity, None, data)
+
+        PrivacyRequestRunner(
+            cache=cache,
+            privacy_request=privacy_request,
+        ).submit()
+
+        return PrivacyRequestDRPStatusResponse(
+            request_id=privacy_request.id,
+            received_at=privacy_request.requested_at,
+            status=DrpFidesopsMapper.map_status(privacy_request.status),
+        )
+
+    except common_exceptions.RedisConnectionError as exc:
+        logger.error("RedisConnectionError: %s", exc)
+        # Thrown when cache.ping() fails on cache connection retrieval
+        raise HTTPException(
+            status_code=HTTP_424_FAILED_DEPENDENCY,
+            detail=exc.args[0],
+        )
+    except Exception as exc:
```

Comment:
These `except` blocks should surround only the minimal code, and we should avoid using broad excepts unless there's a specific reason?

Reply:
this was to be consistent with what we have for POST `"/privacy-request"`. But I can remove it, just lemme know your preference